main/src/routes/spotifyRouter.js

```diff
@@ -43,57 +31,20 @@ router.get("/", function (req, res) {
  * Handles GET request for spotify/login page
  */
 router.get("/login", (req, res) => {
-    let main = req.app.locals.main;
-
-    if (!main || !main.spotifyApi) {
-        req.app.locals.main = new Main();
-        main = req.app.locals.main;
-    }
-
-    const spotifyApi = main.spotifyApi;
     const html = spotifyApi.createAuthorizeURL(scopes);
-    console.log(html);
-
-    const state = generateRandomString(16);
-    res.cookie(stateKey, state);
-    res.redirect(html + "&state=" + state + "&show_dialog=true");
+    res.redirect(html + "&show_dialog=true");
 });
 
 /**
  * Handles GET request for spotify/callback page
  */
 router.get("/callback", async (req, res) => {
-    const state = req.query.state || null;
-    const storedState = req.cookies ? req.cookies[stateKey] : null;
-
-    if (state === null || state !== storedState) {
-        res.redirect("/#" +
-            qs.stringify({
-                error: "state_mismatch",
-            }));
-        return;
-    }
-
-    let main = req.app.locals.main;
-
-    if (!main || !main.spotifyApi || !main.spotify) {
-        req.app.locals.main = new Main();
-        main = req.app.locals.main;
-    }
-
-    const spotifyApi = main.spotifyApi;
     const {code} = req.query;
-    console.log(code);
-
     return spotifyApi.authorizationCodeGrant(code)
         .then((data) => {
-            const {access_token, refresh_token} = data.body;
-            spotifyApi.setAccessToken(access_token);
-            spotifyApi.setRefreshToken(refresh_token);
-            console.log("SPOTIFY - set access and refresh tokens");
-            console.log("ACCESS TOKEN: " + spotifyApi.getAccessToken());
-            console.log("\n");
-            console.log("REFRESH TOKEN: " + spotifyApi.getRefreshToken());
+            const {access_token} = data.body;
+            // only store access_token
+            req.session.access_token = access_token;
```

Comment:
Do we need to edit the Spotify Service to access req.session.access_token?

Reply:
Hmmm I believe not, currently the only services that touch `req.session.access_token` are Main and RefreshCredentialService

Reply:
Oooo I see! How come we set the access token for spotifyApi in main twice both in trackRouter and webcamRouter's POST / requests?

Reply:
Oh because we need to create a new Main each time (since we can't access the same Main from another request) and since main creates a new spotify Api, we need to set the access token for the spotifyApi. 

Essentially, we maintain zero state on our side and delegate the session to maintain state.

Reply:
I see :) thanks for explaining!